packages/hooks-react/src/useProtectedMedia.ts

```diff
@@ -7,5 +8,20 @@ import useContentProtection from './useContentProtection';
 export default function useProtectedMedia(item: PlaylistItem) {
   const apiService = getModule(ApiService);
 
-  return useContentProtection('media', item.mediaid, (token, drmPolicyId) => apiService.getMediaById(item.mediaid, token, drmPolicyId));
+  const [isGeoBlocked, setIsGeoBlocked] = useState(false);
+  const queryResult = useContentProtection('media', item.mediaid, (token, drmPolicyId) => apiService.getMediaById(item.mediaid, token, drmPolicyId));
+
+  useEffect(() => {
+    const m3u8 = queryResult.data?.sources.find((source) => source.file.indexOf('.m3u8') !== -1);
```

Comment:
Maybe add an early return on `!contentProtectionQuery.isSuccess`

Reply:
I think this is not needed, because queryResult.data will not change it's value and therefore the effect will not be called. Also added that the optional chaining will cause the `fetch` to be skipped ;)